src/core/include/openvino/pass/serialize.hpp

```diff
@@ -35,13 +35,21 @@ class OPENVINO_API Serialize : public ov::pass::ModelPass {
 
     Serialize(const std::string& xmlPath, const std::string& binPath, Version version = Version::UNSPECIFIED);
 
+#if defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT)
```

Comment:
discussed offline 

in some places we use "defined(_WIN32)"  and "defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT)" together, 
but in some other places we use only  "defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT)" 

for functions or methods I believe we need to use  "defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT)" only
as it's in  https://github.com/openvinotoolkit/openvino/blob/ac2f7ef07355996e4402405443323893b5dc3484/src/inference/include/openvino/runtime/core.hpp#L82

Reply:
but actually we do not duplicate the code and use string implementation converting wstring to string:
<img width="331" alt="image" src="https://github.com/openvinotoolkit/openvino/assets/21181928/4bc83c83-57e4-4afe-aea4-25f66395cd8c">
and convert wstring to string back inside  (+ "defined(_WIN32)")
<img width="280" alt="image" src="https://github.com/openvinotoolkit/openvino/assets/21181928/97c972ca-e422-46ee-9b1e-cdf7eef67355">

Reply:
I believe we should probably use the same approach as for read_model and try to avoid code duplication

Reply:
Done, please take a look.